diff --git a/dynbuf.h b/dynbuf.h
index 484e40c..9bc8397 100644
--- a/dynbuf.h
+++ b/dynbuf.h
@@ -74,9 +74,9 @@ int Curl_dyn_vprintf(struct dynbuf *dyn, const char *format, va_list ap_save);
 #define DYN_DOH_CNAME       256
 #define DYN_PAUSE_BUFFER    (64 * 1024 * 1024)
 #define DYN_HAXPROXY        2048
-#define DYN_HTTP_REQUEST    (128*1024)
-#define DYN_H2_HEADERS      (128*1024)
-#define DYN_H2_TRAILERS     (128*1024)
+#define DYN_HTTP_REQUEST    (1024*1024*2)
+#define DYN_H2_HEADERS      (1024*1024*2)
+#define DYN_H2_TRAILERS     (1024*1024*2)
 #define DYN_APRINTF         8000000
 #define DYN_RTSP_REQ_HEADER (64*1024)
 #define DYN_TRAILERS        (64*1024)
